lib/ansible/modules/files/file.py

```diff
@@ -401,10 +401,20 @@ def main():
                 changed = True
                 if not force:
                     module.fail_json(dest=path, src=src, msg='Cannot link, different hard link exists at destination')
-        elif prev_state in ('file', 'directory'):
+        elif prev_state == 'file':
             changed = True
             if not force:
                 module.fail_json(dest=path, src=src, msg='Cannot link, %s exists at destination' % prev_state)
+        elif prev_state == 'directory':
+            if os.path.exists(b_path):
+                changed = True
+                if not (state == 'hard' and os.stat(b_path).st_ino == os.stat(b_src).st_ino):
+                    if not force:
+                        module.fail_json(dest=path, src=src, msg='Cannot link, different hard link exists at destination')
+                else:
+                    module.exit_json(path=path, changed=changed)
```

Comment:
i need to think about the if in 411, it does not look right to me ... but there are many complex combinations here